README.md

```diff
@@ -20,34 +20,65 @@ yarn manypkg check
 yarn manypkg fix
 ```
 
-### Script Runner
+## Dictionary
 
-> Note: The following is not implemented currently
+- **private package** - A package that has `private: true`/is not published. It does not refer to a package published to a private registry here.
+- **internal package/workspace** - A package that is local/in the repo
+- **external package** - A package that is from a registry like npm
+- **range** - A [node-semver range](https://github.com/npm/node-semver#ranges)
 
-### In the root of a multi-package repo
+## Checks
 
-```
-yarn m <pkgName> add <depDescriptor>
-```
+## External mismatch
 
-```
-yarn m <pkgName> <scriptName>
-```
+The ranges for all dependencies(excluding `peerDependencies`) on external packages should be equal(`===`). This is so that only a single version of an external package will be installed because having multiple versions of the same package can cause confusion and bundle size problems especially with libraries like React that require there to only be a single copy of the library. It's important to note that this check does not enforce that only a single version of an external package is installed, only that two versions of an external package will never be installed because they're specified as dependencies of internal packages.
 
-```
-yarn m <binaryFromNodeModulesName>
-```
+There are vague thoughts that it should be possible to partially disable external mismatch rules but we have not yet had a use case for it so we have not addressed it yet.
 
-### In a package
+### How it's fixed
 
-```
-yarn m add <depDescriptor>
-```
+The highest range of the dependency is found as defined by [sembear](https://github.com/mitchellhamilton/sembear) and set as the range at every non-peer dependency place it is depended on.
 
-```
-yarn m <scriptName>
-```
+## Internal mismatch
 
-```
-yarn m <binaryFromNodeModulesName>
-```
+The ranges for all dependencies(excluding `peerDependencies`) on internal packages should include the version of the internal package. This is so that an internal package will never depend on another internal package but get the package from the registry because that happening is very confusing and you should always prefer a local version of any given package.
+
+### How it's fixed
+
+If the range is a [caret range](https://github.com/npm/node-semver#caret-ranges-123-025-004) or a [tilde range](https://github.com/npm/node-semver#tilde-ranges-123-12-1) with no other comparators, the range is set as a caret or tilde range respectively with the version of the internal package. If it is any other range, the range is set to the exact version of the internal package.
```

Comment:
> If it is any other range, the range is set to the exact version of the internal package.

This sentence needs more clarity - I *think* I know what this means (It will move `>=10.0.1` to `>=10.0.2` when the package gets patched) but I'm not sure if that's what it means.